### PR TITLE
Dynamo export refactor

### PR DESCRIPTION
**Pull request by peri044**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/backend/test/utils.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 from torch._functorch.aot_autograd import aot_module_simplified, make_boxed_compiler
 
+DECIMALS_OF_AGREEMENT = 4
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/backend/test/utils.py:16]: Would this constant not be necessary/helpful in the `export` testing? As in - could it instead go in `py/torch_tensorrt/dynamo/utils.py`?

File: py/torch_tensorrt/dynamo/utils.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/utils.py:1]: Should `dynamo/utils.py` be distinct from `dynamo/test_utils.py`? I don't think it makes a huge difference here, but if the number of utilities were to increase, it might help to differentiate compilation utilities from testing utilities.

**Review by gs-olive — COMMENTED** [review]
Overall, the changes looks good - made some general organizational comments.

File: py/torch_tensorrt/dynamo/backend/test/utils.py (hunk 1 above)
Comment (peri044) [reply py/torch_tensorrt/dynamo/backend/test/utils.py:16]: This parameter is used by tests which are currently written for `backend`. I don't have equivalent tests yet for export. I will work on those once this is evened out. After, that we can migrate this argument accordingly so that both can use it from a common place.

File: py/torch_tensorrt/dynamo/utils.py
Comment (peri044) [reply py/torch_tensorrt/dynamo/utils.py:1]: Since there's only `cosine_sim` function in `test_utils.py`, I merged it with `dynamo/utils.py`. I'm fine with leaving it in a separate `test_utils.py` if there are more to come. In general, this is what I'm thinking 
We have a single `dynamo/test` directory which has something like 
1) `models`  subdirectory - has test_models.py  - tests 5 standard models with both `export` and `backend`
2) `export` sub directory - has tests specific to `export` workflow (maybe lowering and partitioning)
3) `backend` sub directory - has tests specific to `backend` workflow (maybe lowering and partitioning)

If lowering and partitioning are shared by `export` and `backend`, we could just have scripts instead of directories. But I think there might be other tests specific to these workflows which might be placed in their subdirectories.

File: py/torch_tensorrt/dynamo/__init__.py [hunk 2]
```diff
@@ -1,2 +1,5 @@
+from ._TorchTensorRTModule import TorchTensorRTModule
 from torch_tensorrt.dynamo import fx_ts_compat
-from .backend import compile
+from ._settings import *
+from torch_tensorrt.dynamo import backend
+from torch_tensorrt.dynamo.export import compile
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/__init__.py:5]: Maybe we could avoid directly promoting the `export.compile` function to the top level, as it could become quite confusing that `torch_tensorrt.dynamo.compile` invokes the `torch.export` path. The suggestion made here https://github.com/pytorch/TensorRT/discussions/1826#discussioncomment-6348562 could be helpful for an alternative.
